routers/api/v1/repo/label.go

```diff
@@ -136,6 +136,10 @@ func CreateLabel(ctx *context.APIContext, form api.CreateLabelOption) {
 	//   "201":
 	//     "$ref": "#/responses/Label"
 
+	if len(form.Color) == 6 {
```

Comment:
`models.NewLabel()` doesn't perform any kind of validation on the format of the label. I'd validate the format with a RegExp (`^#[0-9a-fA-F]{6}$`) (either here or at `models.NewLabel()`).